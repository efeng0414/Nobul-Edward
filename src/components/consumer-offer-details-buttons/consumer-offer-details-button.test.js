import React from "react";
import renderer from "react-test-renderer";
import { IntlProvider, intlShape } from "react-intl";
import ConsumerOfferDetailsButtons from "./index";
import { Button } from "antd/lib/radio";

describe("ConsumerOfferDetailsButtons Component", () => {
  const componentProps = {
    intl: global.getIntl(),
    buttonClickHandle: () => {}
  };

  const componentJSX = <ConsumerOfferDetailsButtons {...componentProps} />;

  describe("Component Snapshot", () => {
    const snapshotJSX = <IntlProvider {...global.getIntl()}>{componentJSX}</IntlProvider>;

    it("should match stored snapshot", () => {
      const componentSnapshot = renderer.create(snapshotJSX).toJSON();

      expect(componentSnapshot).toMatchSnapshot();
    });
  });

  describe("Component render", () => {
    const functionsToReset = ["clickHandle"];
    const componentPrototypeResets = functionsToReset.reduce(
      global.createComponentPrototypeResets({ component: ConsumerOfferDetailsButtons }),
      {}
    );
    const componentMount = global.Enzyme.mount(
      componentJSX,
      global.getIntlContext({ intl: componentProps.intl })
    );

    beforeEach(() => {
      functionsToReset.map(
        global.resetPropertyComponentProperty({
          component: ConsumerOfferDetailsButtons,
          componentMount,
          componentPrototypeResets
        })
      );
    });
    describe("OnClick event", () => {
      const mockClickHandle = jest.fn();
      const mockButtonClickHandle = jest.fn();

      it("should call clickHandle on Button click with action Accept offer", () => {
        componentMount.instance().clickHandle = mockClickHandle;
        componentMount.setProps({ a: 1 });
        componentMount
          .find("Button")
          .at(0)
          .simulate("click");
        expect(mockClickHandle).toBeCalled();
      });

      it("should call clickHandle on Button click with action Reject offer", () => {
        componentMount.instance().clickHandle = mockClickHandle;
        componentMount.setProps({ b: 1 });
        componentMount
          .find("Button")
          .at(1)
          .simulate("click");
        expect(mockClickHandle).toBeCalled();
      });
      it("should call buttonClickHandle function with action={REJECT_OFFER} passed when button is clicked", () => {
        componentMount.setProps({ buttonClickHandle: mockButtonClickHandle });
        const action = componentMount
          .find("Button")
          .at(0)
          .getDOMNode()
          .getAttribute("action");
        componentMount
          .find("Button")
          .at(0)
          .simulate("click");
        expect(mockButtonClickHandle).toBeCalledWith({ action });
      });

      it("should call buttonClickHandle function with action={ACCEPT_OFFER} passed when button is clicked", () => {
        componentMount.setProps({ buttonClickHandle: mockButtonClickHandle });
        const action = componentMount
          .find("Button")
          .at(1)
          .getDOMNode()
          .getAttribute("action");
        componentMount
          .find("Button")
          .at(1)
          .simulate("click");
        expect(mockButtonClickHandle).toBeCalledWith({ action });
      });
    });
  });
});
