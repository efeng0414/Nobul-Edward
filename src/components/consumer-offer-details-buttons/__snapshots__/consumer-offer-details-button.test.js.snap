// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConsumerOfferDetailsButtons Component Component Snapshot should match stored snapshot 1`] = `
<div
  className="accept-decline-buttons"
>
  <button
    action="reject"
    className="ant-btn decline-button ant-btn-lg"
    onClick={[Function]}
    type="button"
  >
    <span>
      Decline Proposal
    </span>
  </button>
  <button
    action="accept"
    className="ant-btn accept-button ant-btn-primary ant-btn-lg"
    onClick={[Function]}
    type="button"
  >
    <span>
      Accept Proposal
    </span>
  </button>
</div>
`;
